test/recurly/BaseClient.test.js

```diff
@@ -80,6 +80,28 @@ describe('BaseClient', () => {
     })
   })
 
+  describe('#_buildPath', () => {
+    it('Should return path when options has no params key', () => {
+      const path = '/resources/myid'
+      assert.equal(client._buildPath(path), path)
+    })
+
+    it('Should return path when options.params is an empty object', () => {
+      const path = '/resources/myid'
+      assert.equal(client._buildPath(path, { params: {} }), path)
+    })
+
+    it('Should add params to query string', () => {
```

Comment:
What is the behavior if someone doesn't read the upgrade notes? Does it throw an exception? Do we capture that in tests if it does?

Reply:
```
WARNING: Published ports are discarded when using host network mode
/app/scripts/stubs/node_stub.js:4
process.on('unhandledRejection', up => { throw up })
                                         ^

ApiError [RecurlyApiError]: 'limit' is not a valid option: [body, params]
    at /app/clients/node/lib/recurly/BaseClient.js:75:15
    at Array.forEach (<anonymous>)
    at Client._validateOptions (/app/clients/node/lib/recurly/BaseClient.js:73:26)
    at Client._makeRequest (/app/clients/node/lib/recurly/BaseClient.js:81:10)
    at /app/clients/node/lib/recurly/Pager.js:69:23
    at new Promise (<anonymous>)
    at Object.next (/app/clients/node/lib/recurly/Pager.js:65:16)
    at /app/clients/node/lib/recurly/Pager.js:94:22
    at new Promise (<anonymous>)
    at Object.next (/app/clients/node/lib/recurly/Pager.js:93:18) {
  name: 'RecurlyApiError',
  type: undefined
}
```

assuming this is captured in another test

Reply:
The client throws the invalid options error:
```
ApiError [RecurlyApiError]: 'limit' is not a valid option: [body, params]
```